robolectric/src/test/java/org/robolectric/shadows/ShadowParcelFileDescriptorTest.java

```diff
@@ -255,4 +264,15 @@ public void testGetFd_canRead() throws IOException {
     assertThat(is.read()).isEqualTo(READ_ONLY_FILE_CONTENTS);
     is.close();
   }
+
+  @Test(expected = IllegalStateException.class)
+  public void testGetFd_alreadyClosed() throws Exception {
+    pfd = ParcelFileDescriptor.open(file, ParcelFileDescriptor.MODE_READ_WRITE);
+    assertThat(pfd).isNotNull();
+    assertThat(pfd.getFileDescriptor().valid()).isTrue();
+
+    pfd.close();
+
+    int unused = pfd.getFd();
```

Comment:
nit: we have a preference for `assertThrows`, e.g.:

```java
assertThrows(IllegalStateException.class, () -> pfd.getFd());
```

This way it is slightly more precise and we can check a specific line.

Reply:
This tip is also useful for my current working PR :).

Reply:
> nit: we have a slight preference for `assertThrows`, e.g.:
> 
> ```java
> assertThrows(IllegalStateException.class, () -> pfd.getFd());
> ```
> 
> This way it is slightly more precise and we can check a specific line.

Yeah, assertThrows is new in JUnit 4.13/5, I will follow that.

I think Truth ExpectFailure is another choice as well if we have the preference.

Reply:
Oh nice either assertThrows or expectFailure will work then.